cranelift/codegen/src/isa/pulley_shared/inst/emit.rs

```diff
@@ -224,9 +224,9 @@ fn pulley_emit<P>(
             }
 
             if let Some(try_call) = info.try_call_info.as_ref() {
-                sink.add_call_site(&try_call.exception_dests);
+                sink.add_call_site(try_call.exception_handlers(&state.frame_layout));
             } else {
-                sink.add_call_site(&[]);
+                sink.add_call_site(core::iter::empty());
             }
```

Comment:
Maybe makes sense to define `add_call_site` such that it doesn't take any handlers, and then a new `add_try_call_site` that does take them? Would cut down on the empty iterator argument stuff to clean up all these call sites a little.

Actually, even better would be if we pushed the try call info into the caller, so that call sites just looked like

```rust
sink.add_call_site(info.try_call_info())
```

and then `add_call_site` would internally do the `if let Some(...)`.

Reply:
Went with the former option -- interpreting the `TryCallInfo` to get handlers requires a `FrameLayout` as well, and pushing that into the `MachBuffer` felt like too much coupling (vs. taking an iterator over `MachExceptionHandler`s, which are a buffer-defined concept).